server-common/src/main/java/org/apache/kafka/server/common/TransactionVersion.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Collections;
+import java.util.Map;
+
+public enum TransactionVersion implements FeatureVersion {
+
+    // Version 1 enables flexible transactional state records. (KIP-890)
+    TV_1(1, MetadataVersion.IBP_4_0_IV1, Collections.emptyMap()),
+    // Version 2 enables epoch bump per transaction and optimizations. (KIP-890)
+    TV_2(2, MetadataVersion.IBP_4_0_IV1, Collections.emptyMap());
+
+    public static final String FEATURE_NAME = "transaction.version";
+
+    private final short featureLevel;
+    private final MetadataVersion bootstrapMetadataVersion;
+    private final Map<String, Short> dependencies;
+
+    TransactionVersion(
+            int featureLevel,
+            MetadataVersion bootstrapMetadataVersion,
+            Map<String, Short> dependencies
```

Comment:
nit: Indentation seems to be incorrect here.

Reply:
I need to figure out how to fix this in my IDE 😵